batchconsumer/sender.go

```diff
@@ -0,0 +1,46 @@
+package batchconsumer
+
+import (
+	"errors"
+	"fmt"
+)
+
+// ErrLogIgnored should be returned by EncodeLog when it encounters a log line that will not be
+// consumed
+var ErrLogIgnored = errors.New("Log intentionally skipped by sender")
+
+// Sender an interface needed for batch consumer implementations
+type Sender interface {
+	// EncodeLog receives a raw log line.  It's expected to return an appropriately formated log
+	// as well as a list of tags for that log line.  A tag corresponds to a batch that it'll be
+	// put into.  Typically tags are series names.
+	// If a log line will not be consumed, EncodeLog should return a ErrLogIgnored error.
+	EncodeLog(rawlog []byte) (log []byte, tags []string, err error)
```

Comment:
nit: kinesis terminology calls stream contents "records"... not sure how generic we want to make this, but the language assuming log data in the stream might not be ideal